### ISSUE TITLE
Event 6008

### ISSUE THREAD

**Issue by seldur** [sarog/PSWindowsUpdate#1]
It seems that "shutdown.exe ... /f ..." after installation triggers an event ID 6008 (The system has rebooted without cleanly shutting down first). Is it possible to use a "cleaner" method for restart signal?

Edit:
I'm writing here, because author doesn't have issues enabled.

**Comment by sarog** [comment]
Agreed, there's no need to force a reboot. We can either remove the `/f` flag or replace the command with a proper Powershell equivalent (`Restart-Computer`).

**Comment by sarog** [comment]
I did some research on this issue. According to `shutdown.exe`, any values of `-t` that are greater than 0 imply a forceful shutdown. That is, `-f` is implied when `-t` is greater than 0. Currently, PSWU has enforced `-t 5` for regular reboots. I will change this in my upcoming revision.

Adding `-dp 0:0` seems to avoid the unexpected shutdown dialog by giving it a reason of "Planned: Other".

A potential solution for Scheduled Reboots is to create a scheduled task at a specified time instead of passing the schedule to `shutdown.exe` via `-t`. The scheduled task will simply have a regular reboot command that signals the OS for a graceful shutdown (by default, 2 minutes).

I have already revised both `RunReboot()` and `ScheduleReboot()` functions to include a `forceReboot` parameter (not pushed yet) so the sysadmin can decide via CLI.

Alternatively, there are potential registry-based workarounds, but these are out of the scope of PSWU:
- WaitToKillServiceTimeout
- AutoEndTasks
- HungAppTimeout

**Comment by rokejulianlockhart** [comment]
Definitely replace with Restart-Computer.
